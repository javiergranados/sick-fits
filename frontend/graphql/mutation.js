import gql from 'graphql-tag';

const CREATE_ITEM = gql`
  mutation createItem($title: String!, $description: String!, $price: Int!, $image: String, $largeImage: String) {
    createItem(title: $title, description: $description, price: $price, image: $image, largeImage: $largeImage) {
      id
    }
  }
`;

const UPDATE_ITEM = gql`
  mutation updateItem($id: ID!, $title: String, $description: String, $price: Int) {
    updateItem(id: $id, title: $title, description: $description, price: $price) {
      id
      title
      description
      price
    }
  }
`;

const DELETE_ITEM = gql`
  mutation deleteItem($id: ID!) {
    deleteItem(id: $id) {
      id
      title
    }
  }
`;

const SIGN_UP = gql`
  mutation signup($email: String!, $name: String!, $password: String!) {
    signup(email: $email, name: $name, password: $password) {
      id
      email
      name
    }
  }
`;

const SIGN_IN = gql`
  mutation signin($email: String!, $password: String!) {
    signin(email: $email, password: $password) {
      id
      email
      name
      password
    }
  }
`;

const SIGN_OUT = gql`
  mutation signout {
    signout {
      message
    }
  }
`;

const REQUEST_RESET = gql`
  mutation requestReset($email: String!) {
    requestReset(email: $email) {
      message
    }
  }
`;

export { CREATE_ITEM, UPDATE_ITEM, DELETE_ITEM, SIGN_UP, SIGN_IN, SIGN_OUT, REQUEST_RESET };
