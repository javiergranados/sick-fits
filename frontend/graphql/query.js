import gql from 'graphql-tag';
import { perPage } from '../config';

const GET_ITEMS = gql`
  query($skip: Int = 0, $first: Int = ${perPage}) {
    items(skip: $skip, first: $first, orderBy: createdAt_DESC) {
      id
      title
      price
      description
      image
      largeImage
    }
  }
`;

const GET_ITEM = gql`
  query($id: ID!) {
    item(where: { id: $id }) {
      id
      title
      description
      price
      largeImage
    }
  }
`;

const PAGINATION = gql`
  {
    itemsConnection {
      aggregate {
        count
      }
    }
  }
`;

const CURRENT_USER = gql`
  {
    me {
      id
      name
      email
      permissions
    }
  }
`;

const GET_ALL_USERS = gql`
  {
    users {
      id
      name
      email
      permissions
    }
  }
`;

export { GET_ITEMS, GET_ITEM, PAGINATION, CURRENT_USER, GET_ALL_USERS };
