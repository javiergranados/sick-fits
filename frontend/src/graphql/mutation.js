import gql from 'graphql-tag';

const CREATE_ITEM = gql`
  mutation createItem($title: String!, $description: String!, $price: Int!, $image: String, $largeImage: String) {
    createItem(title: $title, description: $description, price: $price, image: $image, largeImage: $largeImage) {
      id
    }
  }
`;

const UPDATE_ITEM = gql`
  mutation updateItem($id: ID!, $title: String, $description: String, $price: Int) {
    updateItem(id: $id, title: $title, description: $description, price: $price) {
      id
      title
      description
      price
    }
  }
`;

const DELETE_ITEM = gql`
  mutation deleteItem($id: ID!) {
    deleteItem(id: $id) {
      id
    }
  }
`;

export { CREATE_ITEM, UPDATE_ITEM, DELETE_ITEM };
