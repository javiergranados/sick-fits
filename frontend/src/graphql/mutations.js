import gql from 'graphql-tag';

const CREATE_ITEM = gql`
  mutation createItem($title: String!, $description: String!, $price: Int!, $image: String, $largeImage: String) {
    createItem(title: $title, description: $description, price: $price, image: $image, largeImage: $largeImage) {
      id
    }
  }
`;

export { CREATE_ITEM };
