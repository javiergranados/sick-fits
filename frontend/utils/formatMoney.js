export default function(amount) {
  const options = {
    style: 'currency',
    currency: 'EUR',
    minimumFractionDigits: 2,
  };
  // if its a whole, dollar amount, leave off the .00
  if (amount % 100 === 0) options.minimumFractionDigits = 0;
  const formatter = new Intl.NumberFormat('it', options);
  return formatter.format(amount / 100);
}
// number.toLocaleString('it', { style: 'currency', currency: 'EUR' }) : '0,00 â‚¬';
