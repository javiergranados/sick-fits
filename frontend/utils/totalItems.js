export default function(cart) {
  return cart.reduce((prev, next) => prev + next.quantity, 0);
}
