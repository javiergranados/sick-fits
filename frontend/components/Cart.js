import React from 'react';
import { useQuery, useMutation } from '@apollo/react-hooks';
import { GET_LOCAL_STATE } from '../graphql/query';
import { TOGGLE_CART } from '../graphql/mutation';
import { Supreme } from './styles/Supreme';
import { CloseButton } from './styles/CloseButton';
import { SickButton } from './styles/SickButton';
import * as S from './styles/Cart';

const Cart = () => {
  const { data } = useQuery(GET_LOCAL_STATE);
  const [toggleCart] = useMutation(TOGGLE_CART);

  return (
    <S.Cart open={data.cartOpen}>
      <header>
        <CloseButton title="close" onClick={toggleCart}>
          &times;
        </CloseButton>
        <Supreme>Your cart</Supreme>
        <p>You Have __ Items in your cart.</p>
      </header>
      <footer>
        <p>$10.10</p>
        <SickButton>Checkout</SickButton>
      </footer>
    </S.Cart>
  );
};

export default Cart;
