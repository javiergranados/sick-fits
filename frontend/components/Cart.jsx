import React from 'react';
import { useQuery, useMutation } from '@apollo/react-hooks';
import User from './User';
import CartItem from './CartItem';
import TakeMyMoney from './TakeMyMoney';
import { GET_LOCAL_STATE } from '../graphql/query';
import { TOGGLE_CART } from '../graphql/mutation';
import { Supreme } from './styles/Supreme';
import { CloseButton } from './styles/CloseButton';
import { SickButton } from './styles/SickButton';
import calcTotalPrice from '../utils/calcTotalPrice';
import * as S from './styles/Cart';
import formatMoney from '../utils/formatMoney';

const Cart = () => {
  const { data } = useQuery(GET_LOCAL_STATE);
  const [toggleCart] = useMutation(TOGGLE_CART);

  return (
    <User>
      {({ me }) => {
        if (!me) {
          return null;
        }
        return (
          <S.Cart open={data.cartOpen}>
            <header>
              <CloseButton title="close" onClick={toggleCart}>
                &times;
              </CloseButton>
              <Supreme>{me.name}'s cart</Supreme>
              <p>
                You Have {me.cart.length} Item{me.cart.length > 1 && 's'} in your cart.
              </p>
            </header>
            <ul>
              {me.cart.map(cartItem => (
                <CartItem key={cartItem.id} cartItem={cartItem} />
              ))}
            </ul>
            <footer>
              <p>{formatMoney(calcTotalPrice(me.cart))}</p>
              <TakeMyMoney>
                <SickButton>Checkout</SickButton>
              </TakeMyMoney>
            </footer>
          </S.Cart>
        );
      }}
    </User>
  );
};

export default Cart;
