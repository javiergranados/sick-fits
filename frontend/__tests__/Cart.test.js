import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from '@apollo/react-testing';
import { act } from 'react-dom/test-utils';
import Cart from '../components/Cart';
import { CURRENT_USER, GET_LOCAL_STATE } from '../graphql/query';
import { fakeUser, fakeCartItem } from '../utils/testUtils';

const user = fakeUser();

const mocks = [
  {
    request: { query: CURRENT_USER },
    result: {
      data: {
        me: {
          ...user,
          cart: [fakeCartItem()],
        },
      },
    },
  },
  {
    request: { query: GET_LOCAL_STATE },
    result: { data: { cartOpen: true } },
  },
];

describe('<Cart />', () => {
  it.skip('renders and matches snappy', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks} client={{}}>
        <Cart />
      </MockedProvider>
    );

    await act(wait);
    wrapper.update();

    expect(toJSON(wrapper.find('header'))).toMatchSnapshot();
    expect(wrapper.find('ul').children()).toHaveLength(1);
  });
});
