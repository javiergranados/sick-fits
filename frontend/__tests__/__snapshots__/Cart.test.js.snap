// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> renders and matches snappy 1`] = `
<header>
  <CloseButton
    onClick={[Function]}
    title="close"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "CloseButton-sc-1seb878-0",
            "isStatic": false,
            "lastClassName": "bgimpA",
            "rules": Array [
              "background:black;color:white;font-size:3rem;border:0;position:absolute;z-index:2;right:0;cursor:pointer;",
            ],
          },
          "displayName": "CloseButton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "CloseButton-sc-1seb878-0",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      title="close"
    >
      <button
        className="CloseButton-sc-1seb878-0 bgimpA"
        onClick={[Function]}
        title="close"
      >
        Ã—
      </button>
    </StyledComponent>
  </CloseButton>
  <Supreme>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Supreme-xv30qb-0",
            "isStatic": false,
            "lastClassName": "hpaXsq",
            "rules": Array [
              "background:",
              [Function],
              ";color:white;display:inline-block;padding:4px 5px;transform:skew(-3deg);margin:0;font-size:4rem;",
            ],
          },
          "displayName": "Supreme",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Supreme-xv30qb-0",
          "target": "h3",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h3
        className="Supreme-xv30qb-0 hpaXsq"
      >
        Miss Coleman Berge
        's cart
      </h3>
    </StyledComponent>
  </Supreme>
  <p>
    You Have 
    1
     Item
     in your cart.
  </p>
</header>
`;
